public/docs/dart/latest/tutorial/toh-pt5.jade

```diff
@@ -1,14 +1,696 @@
+
++makeExample('toh-5/dart/lib/app_component.dart','template', 'app_component.dart (template)')(format=".")
+.l-sub-section
+    :marked
+        We nestled the two links within `<nav>` tags.
+        They don't do anything yet but they'll be convenient when we style the links a little later in the chapter.
+:marked
+    Refresh the browser. The app displays the dashboard and
+    we can navigate between the dashboard and the heroes.
+
+    ## Dashboard Top Heroes
+    Let’s spice up the dashboard by displaying the top four heroes at a glance.
+
+    Replace the `template` metadata with a `templateUrl` property that points to a new
+    template file.
+
++makeExample('toh-5/dart/lib/dashboard_component.dart', 'template-url', 'lib/dashboard_component.dart (templateUrl)')(format=".")
+.l-sub-section
+    :marked
+        We specify the path _all the way back to the application root_. Angular doesn't support module-relative paths.
+:marked
+    Create that file with these contents:
++makeExample('toh-5/dart/lib/dashboard_component.html', null, 'dashboard_component.html')(format=".")
+:marked
+    We use `*ngFor` once again to iterate over a list of heroes and display their names.
+    We added extra `<div>` elements to help with styling later in this chapter.
+
+    There's a `(click)` binding to a `gotoDetail` method we haven't written yet and
+    we're displaying a list of heroes that we don't have.
+    We have work to do, starting with those heroes.
+
+    ### Share the *HeroService*
+
+    We'd like to re-use the `HeroService` to populate the component's `heroes` list.
+
+    Recall earlier in the chapter that we removed the `HeroService` from the `providers` list of the `HeroesComponent`
+    and added it to the `providers` list of the top level `AppComponent`.
+
+    That move created a singleton `HeroService` instance, available to *all* components of the application.
+    We'll inject and use it here in the `DashboardComponent` .
+
+    ### Get heroes
+    Open the `dashboard_component.dart` and add the requisite `import` statements.
++makeExample('toh-5/dart/lib/dashboard_component_2.dart','imports', 'lib/dashboard_component.dart (imports)')(format=".")
+:marked
+    We need `OnInit` interface because we'll initialize the heroes in the `ngOnInit` method as we've done before.
+    We need the `Hero` and `HeroService` symbols in order to reference those types.
+
+    Now implement the `DashboardComponent` class like this:
++makeExample('toh-5/dart/lib/dashboard_component_2.dart','component', 'lib/dashboard_component.dart (class)')
+:marked
+    We saw this kind of logic before in the `HeroesComponent`.
+    * create a `heroes` list property
+    * inject the `HeroService` in the constructor and hold it in a private `_heroService` field.
+    * call the service to get heroes inside the Angular `ngOnInit` lifecycle hook.
+
+    The noteworthy differences: we cherry-pick four heroes (2nd, 3rd, 4th, and 5th) with *slice*
```

Comment:
The "with *slice*" suggests there is a `slice` operation in Dart. Maybe drop that fragment or just leave it for now and we can address it on the next round (when the doc is refactored to conform to the new design).